src/controllers/user.js

```diff
@@ -12,7 +14,56 @@ const getProfile = async (req, res) => {
         .json(currentUser);
     }
   } catch (error) {
-    return res.sendStatus(500).json({ message: error.message });
+    return res.status(500).json({ message: error.message });
+  }
+};
+
+const getUserProducts = async (req, res) => {
+  try {
+    let { postType } = req.query;
+    const { search } = req.query;
+    const filter = { $and: [] };
+    const filterQueries = { $or: [] };
+
+    if (postType) {
+      if (typeof postType !== 'object') postType = postType.split();
+
+      postType.forEach((type) => {
+        if (type === 'donated') {
```

Comment:
we can uppercase the first letters for donated, received and requested

Reply:
Sure you're right